### PR TITLE
Add more coding standard rules

### PR DESCRIPTION
**Pull request by fabpot**

### CODE REVIEW

**Comment by stof** [comment]
IIRC, there was a discussion on the Symfony repo reporting that Twig coding standards don't define the spacing rule around the `=` operator in macro argument declarations and around the `=` or `:` separators for named arguments.
Maybe those could be defined as well.

**Comment by VincentLanglet** [comment]
> IIRC, there was a discussion on the Symfony repo reporting that Twig coding standards don't define the spacing rule around the `=` operator in macro argument declarations and around the `=` or `:` separators for named arguments. Maybe those could be defined as well.

For the record, currently Twig-CS-Fixer consider it should be 
```
foo(name_arg=42) // no space at all
```
and
```
foo(name_args: 42) // one space after
```
cf
https://github.com/VincentLanglet/Twig-CS-Fixer/blob/main/src/Rules/Function/NamedArgumentSpacingRule.php

**Comment by fabpot** [comment]
Updated

**Review by VincentLanglet — APPROVED** [review]
I'll work on implementing the missing rule in Twig-CS-Fixer :)

File: doc/coding_standards.rst [hunk 1]
```diff
@@ -88,21 +88,34 @@ standards:
...
+* Do not put any spaces before and after ``=`` in macro argument declarations:
 
   .. code-block:: twig
 
-    {% set name = 'Fabien' %}
-    {% set first_name = 'Fabien' %}
+    {% macro html_input(class="input") %}
 
-* Use snake case for all function/filter/test names:
+* Put exactly one space after the ``:`` sign in macro argument declarations:
```
Comment (VincentLanglet) [inline doc/coding_standards.rst:97]: @fabpot It works too for named arguments
```
{{ html_input(class_name: 'pwd') }}
```

File: doc/coding_standards.rst [hunk 2]
```diff
@@ -88,21 +88,34 @@ standards:
     [1, 2, 3]
     {'name': 'Fabien'}
 
-* Use snake case for all variable names (provided by the application and
-  created in templates):
+* Do not put any spaces before and after ``=`` in macro argument declarations:
```
Comment (VincentLanglet) [inline doc/coding_standards.rst:91]: It works too for named arguments
```
{{ html_input(class_name='pwd') }}
```

File: doc/coding_standards.rst (hunk 1 above)
Comment (VincentLanglet) [reply doc/coding_standards.rst:97]: And I just noticed, but I don't think this macro definition is a valid one.
```
{% macro html_input(class: "input") %}
```

File: doc/coding_standards.rst (hunk 2 above)
Comment (VincentLanglet) [reply doc/coding_standards.rst:91]: I was wrong for macro, seems like it should be ` = ` cf https://twig.symfony.com/doc/3.x/tags/macro.html

**Comment by VincentLanglet** [comment]
I opened https://github.com/twigphp/Twig/pull/4488

**Pull request merged**